platform/platform-impl/src/com/intellij/toolWindow/ToolWindowHeader.kt

```diff
@@ -188,6 +188,9 @@ abstract class ToolWindowHeader internal constructor(
           }
           else {
             toolWindow.fireActivated(ToolWindowEventSource.ToolWindowHeader)
+            // Move focus the context component.
+            val decorator = InternalDecoratorImpl.findNearestDecorator(this@ToolWindowHeader)
+            decorator?.requestContentFocus()
```

Comment:
Please consider the mouse listener in `com.intellij.openapi.wm.impl.content.ToolWindowContentUi#initMouseListeners`. 
I'm not sure, but maybe it would be a better place to switch the focused component.

Reply:
This listener is installed on `TabPanel` but we need a listener on the whole header.